x-pack/plugins/infra/public/pages/metrics/settings/source_configuration_settings.tsx

```diff
@@ -75,19 +75,13 @@ export const SourceConfigurationSettings = ({
     formStateChanges,
   ]);
 
-  const isWriteable = useMemo(
-    () => shouldAllowEdit && source && source.origin !== 'internal',
-    [shouldAllowEdit, source]
-  );
+  const isWriteable = shouldAllowEdit && (!Boolean(source) || source?.origin !== 'internal');
```

Comment:
**nit**: I removed the memoization since calculating boolean values without expensive calculations, usually causes a de-optimization. 
I measure to be sure and is actually much slower on the first render and keeps being slower on successive renders.

**First render**
<img width="216" alt="first-render" src="https://user-images.githubusercontent.com/34506779/203764438-848cc355-47c9-447a-879b-263345e64d4c.png">

**Following renders**
<img width="210" alt="following-renders" src="https://user-images.githubusercontent.com/34506779/203764532-b63393d9-eda0-4f55-8cd5-6ab2b53533aa.png">

Reply:
Nice!

Just one question about how we calculate it. Would this be the same? I guess the previous code was from the time when there was no optional chaining available.

```suggestion
  const isWriteable = shouldAllowEdit && source?.origin !== 'internal';
```

Reply:
It won't be the same because, when the API does not return a configuration, we still want to render the empty form to the user.
With the condition written as `shouldAllowEdit && source?.origin !== 'internal';`, if the config is not fetched and is `undefined` the form won't be enabled for changes.
Instead, with the configuration written in that way (as you can see, the first Boolean casting is negated), we will disable the form only if the configuration exists but the origin is different from `internal`.